### PR TITLE
V3

### PR DESCRIPTION
**Pull request by pmmmwh**
The Node.js and TypeScript ecosystems have evolved and so should we 😄

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,11 +24,22 @@
...
-    "typescript": "^3.6.4",
-    "prettier": "^1.18.2"
+    "@types/node": "^14.14.22",
+    "typescript": "^4.1.3",
+    "prettier": "^2.2.1"
+  },
+  "peerDependencies": {
+    "@types/node": "^14.14.22"
+  },
+  "peerDependenciesMeta": {
```
Comment (pmmmwh) [inline package.json:37]: `@types/node` is only relevant for users on TypeScript. Marking it as an optional peer dependency skips its installation for users not on TS.

File: package.json [hunk 2]
```diff
@@ -23,11 +24,22 @@
...
+    "@types/node": {
+      "optional": true
+    }
+  },
+  "engines": {
+    "node": ">=10.13.0"
   }
```
Comment (pmmmwh) [inline package.json:44]: Node.js v10 is the oldest non-EoL release and supports all `ES2018` features

**Review by a7ul — APPROVED** [review]

**Comment by a7ul** [comment]
Maybe also add the github actions from anyfin/acl so that we can auto deploy this ? @pmmmwh

File: src/bankid.ts [hunk 3]
```diff
@@ -150,39 +151,64 @@ interface BankIdClientSettings {
...
+  readonly request?: any;
+
+  constructor(request?: any) {
+    super(REQUEST_FAILED_ERROR);
+    Error.captureStackTrace(this, this.constructor);
+
+    this.name = "RequestError";
+    this.request = request;
+  }
+}
```
Comment (pmmmwh) [inline src/bankid.ts:182]: Custom error classes - so now you can use `instanceof` checks for preferential error handling

**Comment by pmmmwh** [comment]
Added GitHub Actions ⭐

**Comment by a7ul** [comment]
@pmmmwh  final tiny thing

Can you add this into the readme too

```
# Deploy/Publish

In order to deploy new versions, simply bump the version in package.json and create a new github release.

Github action should automagically deploy it to npm. ✨
```

**Pull request merged**